jaxrs-tck/src/main/java/ee/jakarta/tck/ws/rs/jaxrs31/ee/multipart/MultipartSupportIT.java

```diff
@@ -0,0 +1,427 @@
+        }
+    }
+
+    private static EntityPart find(final Collection<EntityPart> parts, final String name) {
+        for (EntityPart part : parts) {
+            if (name.equals(part.getName())) {
+                return part;
+            }
+        }
+        return null;
+    }
+
+    private static URI createCombinedUri(final URI uri, final String path) throws URISyntaxException {
+        if (path == null || path.isEmpty()) {
+            return uri;
+        }
+        String uriString = uri.toString();
+
+        TestUtil.logMsg("Initial uri string: " + uriString);
+
+        final StringBuilder builder = new StringBuilder(uriString);
+        if (builder.charAt(builder.length() - 1) == '/') {
+            if (path.charAt(0) == '/') {
+                builder.append(path.substring(1));
+            } else {
+                builder.append(path);
+            }
+        } else if (path.charAt(0) == '/') {
+            builder.append(path.substring(1));
+        } else {
+            builder.append('/').append(path);
+        }
+
+        String builderString = builder.toString();
+        TestUtil.logMsg("Combined uri string: " + builderString);
+
+        return new URI(builderString);
+    }
+
+    @ApplicationPath("/")
+    public static class MultipartTestApplication extends Application {
+    }
+
+    @Path("/test")
+    public static class TestResource {
+
+        @POST
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/basicTest")
+        public Response basicTest(final List<EntityPart> parts) throws IOException {
+            final List<EntityPart> multipart = List.of(
+                    EntityPart.withName("received-string")
+                            .content(find(parts,"octet-stream").getContent(byte[].class))
+                            .mediaType(MediaType.APPLICATION_OCTET_STREAM_TYPE)
+                            .build(),
+                    EntityPart.withName("received-file")
+                            .content(find(parts,"file").getFileName().get(),find(parts,"file").getContent())
+                            .mediaType(MediaType.APPLICATION_XML)
+                            .build(),
+                    EntityPart.withName("added-input-stream")
+                            .content(new ByteArrayInputStream("Add part on return.".getBytes()))
+                            .mediaType("text/asciidoc")
+                            .header("Header1","Test1")
+                            .header("Header2","Test2")
+                            .build());
+            return Response.ok(new GenericEntity<>(multipart) {
+                    }, MediaType.MULTIPART_FORM_DATA).build();
+        }
+
+        @POST
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        @Produces(MediaType.MULTIPART_FORM_DATA)
+        @Path("/single-form-param")
+        public List<EntityPart> singleParamTest(@FormParam("test-string") final String string,
```

Comment:
The same here: `test-string` EntityPart has content read three times: once for string, once for EntityPart and once for the InputStream. But the javadoc mandates to throw IllegalStateException.

Reply:
I wonder how that's implemented in RestEasy, because there would need to be some automatic buffering of the InputStream, but that's gonna bite you for large data.

Reply:
I believe this is no longer an issue

Reply:
How about taking the "singleParamTest" resource method and splitting it into 3, each with one `@FormParam` (one EntityPart, one String, and one InputStream) and then have the `singleFormParamTest()` invoke each separately, using the same `List<EntityPart>`?

Reply:
Or maybe this test isn't all that useful or needed?

Reply:
Splitting the test into three would be basically the same as the `multiFormParamTest`, would it not?
The important part is whether the spec was meant to support multiple `@FormParam("test-string")`. My understanding is it was not, because the content stream needs to be read (and buffered) multiple times, but perhaps it was meant so for the resources.

Reply:
Yes, not much difference at all.    I'm leaning toward removal of that test and potentially clarifying the spec in this respect for the next release if we choose to go that route.

Reply:
That being said, can we agree that there also is currently not a spec (or javadoc) requirement for an IllegalStateException if user attempts to do something like what is in this test.   It is nebulous... they might get an exception or they might not depending on the the implemenation.

Reply:
I've removed the test.   We can revisit and potentially augment the spec for V 5.0.